Source/DafnyCore/Rewriters/RefinementTransformer.cs

```diff
@@ -881,7 +906,7 @@ public RefinementTransformer(Program p)
                   replacementBody = MergeBlockStmt(replacementBody, prevMethod.BodyForRefinement);
                 }
               }
-              var newM = CloneMethod(prevMethod, m.Ens, decreases, replacementBody, prevMethod.BodyForRefinement == null, m.Attributes);
+              var newM = CloneMethod(prevMethod, m, decreases, replacementBody, prevMethod.BodyForRefinement == null, m.Attributes);
```

Comment:
Here I needed the entire new method, not just its ensures clause, so that I recover correctly the method's token.